src/shims/native_lib/mod.rs

```diff
@@ -1,8 +1,7 @@
 //! Implements calling functions from a native library.
 
-// FIXME: disabled since it fails to build on many targets.
-//#[cfg(target_os = "linux")]
-//pub mod trace;
+#[cfg(trace)]
+pub mod trace;
```

Comment:
To reduce the amount of `cfg` in this file, I would suggest something like this:
```rust
#[cfg_attr(not(all(target_os = "linux", any(target_arch = "x86_64", ...), path = "trace/stub.rs")]
mod trace;
```
and then the `stub.rs` file contains just a bunch of empty functions and dummy type aliases so that to the outside, it looks like all the same functions and types exist.